app/models/piece.rb

```diff
@@ -1,6 +1,7 @@
 class Piece < ActiveRecord::Base
   belongs_to :user
   belongs_to :game
+  after_initialize :set_image
```

Comment:
once a Piece is initialized, it knows its type, so we can associate the correct image